roles/app/tasks/main.yml

```diff
@@ -1,31 +1,37 @@
+---
 
-- name: Check if PHP7 is needed
+- name: Check if any components to be installed need PHP 7.2
   set_fact:
-    needphp7: True
-  when: "'stepup-webauthn' in group_names or 'stepup-azuremfa' in group_names"
+    # A | intersect(B): list of items that exist in both A and B
 
-# Install the "REMI" repo which contain newer php packages that override the default
-# that come with the distro (CentOS7: 5.4; CentOS6: 5.3)
-- name: Install REMI repo
-  yum: name=http://rpms.famillecollet.com/enterprise/remi-release-7.rpm state=present
+    # stepup-webauthn and stepup-azuremfa always require php 7.2, so install php 7.2 when these must be installed
+    # When one of the other components requires php7.2 this is set in app_php72_components
```

Comment:
Why not add webauthn and azuremfa to app_php72_components, instead of writing a hard-coded exception here?